langchain/chains/query_constructor/prompt.py

```diff
@@ -0,0 +1,137 @@
+# flake8: noqa
+from langchain import PromptTemplate
+
+song_data_source = """\
+```json
+{
+    content: "Lyrics of a song",
+    attributes: {
+        "artist": {
+            "type": "string",
+            "description": "Name of the song artist"
+        },
+        "length": {
+            "type": "integer",
+            "description": "Length of the song in seconds"
+        },
+        "genre": {
+            "type": "string",
+            "description": "The song genre, one of \"pop\", \"rock\" or \"rap\""
+        }
+    }
+}
+```\
+""".replace(
+    "{", "{{"
+).replace(
+    "}", "}}"
+)
+
+full_answer = """\
```

Comment:
What do you think about trying to use an XML tag instead of using JSON encoding?

```
<command>
request("teenager love", and(or(eq(...))
</command>
```

To avoid having to deal with JSON. It feels like the JSON encoding could introduce errors due to the need to escape `"`

If it works it'll also unify the handling of query parameter and the filter to be parsed using the specified grammar